### PR TITLE
[DJFS-153] Article Publishing Endpoints

### PR DESCRIPTION
**Pull request by Samju200**
slim-api
## Task: `Name of given task`
- `Changes you made to given task`

### CODE REVIEW

**Review by adaugochi — CHANGES_REQUESTED** [review]
Kindly push your code too

**Review by adaugochi — CHANGES_REQUESTED** [review]
You are not supposed to push your `vendor` folder. Kindly place it in your `.gitignore` file. What you will do now is:
- Delete your `vendor` folder
- Add the `vendor` folder to `.gitignore`
- run `composer install`
- `add`, `commit` and `push`

**Comment by Samju200** [comment]
> You are not supposed to push your `vendor` folder. Kindly place it in your `.gitignore` file. What you will do now is:
> 
> * Delete your `vendor` folder
> * Add the `vendor` folder to `.gitignore`
> * run `composer install`
> * `add`, `commit` and `push`

Done
Thank you @adaugochi

**Review by adaugochi — CHANGES_REQUESTED** [review]
You didn't do it. Always check your PR to know if your fix what you are asked to do. Click on the `File Changes` tab to see if you do what is required of you. 


On your `.gitignore` file add `vendor` not `vendor/`

**Comment by Samju200** [comment]
> You didn't do it. Always check your PR to know if your fix what you are asked to do. Click on the `File Changes` tab to see if you do what is required of you.
> 
> On your `.gitignore` file add `/vendor` not `vendor/`. Also if you are on windows, add `.idea` to your gitignore file. or Mac, add `.DS_Store` and `composer.lock` file

Yes 
my gitignore is not working for folders
I try to ignore other folders too but still not working

**Comment by Samju200** [comment]
When I run git status --ignored it only bring out .env file showing that others folder in the gitignore is not working

**Comment by adaugochi** [comment]
routes and database should be ignored

**Comment by Samju200** [comment]
> routes and database shouldn't be ignored

Yes just trying it out but still not ignore

File: submissions/adeyemiJuwon/slimPHP/article/app/controllers/ArticleController.php [hunk 1]
```diff
@@ -0,0 +1,177 @@
...
+
+class ArticleController{
+
+
+     private $db;
+    public function __construct(){
+        $this->db =(new DB())->connect();
+    }
+
+    public function index(Request $request, Response $response, $args){
```
Comment (adaugochi) [inline submissions/adeyemiJuwon/slimPHP/article/app/controllers/ArticleController.php:22]: This isn't part of the endpoint you are ask to get

**Review by adaugochi — CHANGES_REQUESTED** [review]
Remove your composer.lock file as part of your PR.

Well done so far

File: submissions/adeyemiJuwon/slimPHP/article/app/controllers/ArticleController.php (hunk 1 above)
Comment (Samju200) [reply submissions/adeyemiJuwon/slimPHP/article/app/controllers/ArticleController.php]: Okay
Thank You @adaugochi

File: submissions/adeyemiJuwon/slimPHP/article/app/controllers/ArticleController.php [hunk 2]
```diff
@@ -0,0 +1,164 @@
...
+            if(!$article) {
+            $response->getBody()->write(json_encode(['error' => 'Article not found']));
+            return $response->withHeader('Content-Type', 'application/json')->withStatus(404);
+        } else if ('status' ==='published') {
+            $response->getBody()->write(json_encode(['error' => 'Article already published']));
+            return $response->withHeader('Content-Type', 'application/json')->withStatus(400);
+        }
+        
+             $response->getBody()->write(json_encode(['status' => 'success',
+'data' => $article]));
```
Comment (adaugochi) [inline submissions/adeyemiJuwon/slimPHP/article/app/controllers/ArticleController.php:117]: format this link of code

File: submissions/adeyemiJuwon/slimPHP/article/app/controllers/ArticleController.php [hunk 3]
```diff
@@ -0,0 +1,164 @@
...
+             return $response ->withHeader('Content-Type', 'application/json')->withStatus(400);
+        }
+    }
+    
+     public function publishArticle(Request $request, Response $response, $args){
+        
+        $id = $args['id'];
+        $status = $args['published'];
+
+
```
Comment (adaugochi) [inline submissions/adeyemiJuwon/slimPHP/article/app/controllers/ArticleController.php:97]: Too much new line space. One newline space is too

File: submissions/adeyemiJuwon/slimPHP/article/app/controllers/ArticleController.php [hunk 4]
```diff
@@ -0,0 +1,164 @@
...
+        }catch(PDOException $ex){
+             $response->getBody()->write(json_encode(['error'=>$ex->getMessage()]));
+             return $response ->withHeader('Content-Type', 'application/json')->withStatus(400);
+        }
+    }
+    public function updateArticle(Request $request, Response $response, $args)  {
+       $id = $args['id'];
+       $title = $request->getParsedBody()['title'];
+       $description = $request->getParsedBody()['description'];
+
```
Comment (adaugochi) [inline submissions/adeyemiJuwon/slimPHP/article/app/controllers/ArticleController.php:57]: Too much new line space. One newline space is too

File: submissions/adeyemiJuwon/slimPHP/article/app/controllers/ArticleController.php [hunk 5]
```diff
@@ -0,0 +1,164 @@
...
+    public function createArticle(Request $request, Response $response, $args){
+        
+        $requestData = $request->getParsedBody();
+        $title = $requestData['title'];
+        $description= $requestData['description'];
+        $status = $requestData['status'];
+        $created_by = $requestData['created_by'];
+        $createAt = date('y-m-d H:i:s');
+
+        $sql =  'INSERT INTO articles (title, description, status, created_by, created_at )VALUES(:title, :description, :status,:created_by, :created_at)';
```
Comment (adaugochi) [inline submissions/adeyemiJuwon/slimPHP/article/app/controllers/ArticleController.php:32]: Break this query. it is too long to stay in one line

File: submissions/adeyemiJuwon/slimPHP/article/app/controllers/ArticleController.php [hunk 6]
```diff
@@ -0,0 +1,164 @@
+<?php
+
+namespace App\Controller;
+
+
+use DB\DB;
+use Psr\Http\Message\ResponseInterface as Response;
+use Psr\Http\Message\ServerRequestInterface as Request;
+use PDO;
+use PDOException;
```
Comment (adaugochi) [inline submissions/adeyemiJuwon/slimPHP/article/app/controllers/ArticleController.php:10]: Too much new line space. One newline space is too

File: submissions/adeyemiJuwon/slimPHP/article/app/controllers/ArticleController.php [hunk 7]
```diff
@@ -0,0 +1,164 @@
...
+
+
+
+class ArticleController{
+
+
+     private $db;
```
Comment (adaugochi) [inline submissions/adeyemiJuwon/slimPHP/article/app/controllers/ArticleController.php:17]: Format this line of code

File: submissions/adeyemiJuwon/slimPHP/article/app/controllers/ArticleController.php [hunk 8]
```diff
@@ -0,0 +1,164 @@
...
+    public function __construct(){
+        $this->db =(new DB())->connect();
+    }
+
+
+    public function createArticle(Request $request, Response $response, $args){
+
```
Comment (adaugochi) [inline submissions/adeyemiJuwon/slimPHP/article/app/controllers/ArticleController.php:24]: Remove space

**Review by adaugochi — CHANGES_REQUESTED** [review]

File: submissions/adeyemiJuwon/slimPHP/article/app/controllers/ArticleController.php (hunk 8 above)
Comment (Samju200) [reply submissions/adeyemiJuwon/slimPHP/article/app/controllers/ArticleController.php]: Okay
Done Thank You @adaugochi

**Review by adaugochi — APPROVED** [review]

**Pull request merged**